src/modules/dns.c

```diff
@@ -667,10 +669,12 @@ static void decode_rr(struct dns_check_info *ci, struct dns_parse *p,
   else
     *output = strdup(txt_str);
   ci->nrr++;
+  if (txt_str_malloc) { free(txt_str_malloc); }
```

Comment:
Don't need the `if` on this. If `txt_str_malloc` is null, free will behave properly (IE: It will do nothing)